### ISSUE TITLE
Combine this package with SparseMatricesCOO

### ISSUE THREAD

**Issue by tmigot** [JuliaSmoothOptimizers/NLPModelsJuMP.jl#109]
We could probably simplify a few things using https://github.com/JuliaSmoothOptimizers/SparseMatricesCOO.jl

- [ ] `SparseMatrixCOO` instead of `COO` https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/blob/c05a30f503d990823407159492fd40b27fc2ee36/src/utils.jl#L28
- [ ] Move `coo_sym_dot` https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/blob/c05a30f503d990823407159492fd40b27fc2ee36/src/utils.jl#L74 to https://github.com/JuliaSmoothOptimizers/SparseMatricesCOO.jl/blob/main/src/coo_linalg.jl

**Comment by amontoison** [comment]
It's an excellent idea @tmigot, I didn't find the time to look at `SparseMatricesCOO.jl` but we should use it as a dependency of `NLPModelsJuMP.jl`.

**Comment by dpo** [comment]
I think it should be a dependency of NLPModels to begin with.

**Comment by tmigot** [comment]
Agreed, this is connected to https://github.com/JuliaSmoothOptimizers/NLPModels.jl/issues/389
